clients/go/pkg/worker/jobWorker_builder.go

```diff
@@ -190,7 +193,7 @@ func (builder *JobWorkerBuilder) Open() JobWorker {
 	}
 }
 
-func NewJobWorkerBuilder(gatewayClient pb.GatewayClient, jobClient JobClient) JobWorkerBuilderStep1 {
+func NewJobWorkerBuilder(gatewayClient pb.GatewayClient, jobClient JobClient, retryPred func(ctx context.Context, err error) bool) JobWorkerBuilderStep1 {
```

Comment:
Just realized, we don't document the public API in this package? :man_shrugging: 

Do you think it would make sense to pass here the `CredentialsProvider` instead? It might make the intent of the predicate clearer, and we could document it as "you should pass the same credentials provider used when creating the client" or something.